src/core/automount.c

```diff
@@ -507,7 +507,12 @@ static void automount_trigger_notify(Unit *u, Unit *other) {
         assert(other);
 
         /* Filter out invocations with bogus state */
-        if (other->load_state != UNIT_LOADED || other->type != UNIT_MOUNT)
+        if (!IN_SET(other->load_state,
+                    UNIT_LOADED,
+                    UNIT_NOT_FOUND,
+                    UNIT_BAD_SETTING,
+                    UNIT_ERROR,
+                    UNIT_MASKED)|| other->type != UNIT_MOUNT)
```

Comment:
Hmm, the states that are left are UNIT_STUB and UNIT_MERGED. I think it'd be cleaner at this point to invert the condition (and maybe even emit a warning if we hit that path. Wouldn't that mean some fuckup in the unit handling logic if a stub unit or a merged unit emitted a notification?).

Also, space before `||` is missing.